File: testers/avro/java-async/generated-and-checked-in/com/example/events/consumer/TreeNodeHandler.java
package com.example.events.consumer;

import com.example.events.TreeNode;
import com.example.events.header.StandardHeaders;
import java.util.concurrent.CompletableFuture;

/** Handler interface for tree-node topic events */
public interface TreeNodeHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(String key, TreeNode value, StandardHeaders headers);
}

File: testers/avro/kotlin/generated-and-checked-in/com/example/events/consumer/TreeNodeHandler.kt
package com.example.events.consumer

import com.example.events.TreeNode
import com.example.events.header.StandardHeaders

/** Handler interface for tree-node topic events */
interface TreeNodeHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: TreeNode,
    headers: StandardHeaders
  )
}